### PR TITLE
[Robot] Create an open recurring donation with a manual opportunity

### PR DESCRIPTION
**Pull request by rjanjanam**
- Usecase to create a new Recurring donation with manual opportunity. Edit day of the month and ensure the details of the opportunities match the expected values
- Fixes for current year and next year calculations
- Keyword doc update

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: robot/Cumulus/resources/RecurringDonationsPageObject.py [hunk 1]
```diff
@@ -198,6 +209,35 @@ def get_next_payment_date_number(self, paynum):
...
+            datefield = npsp_lex_locators["erd"]["installment_date"].format(i)
+            installment_date = self.selenium.get_webelement(datefield)
+            actual_date = self.selenium.get_webelement(installment_date).text
+            year = datetime.strptime(actual_date, "%m/%d/%Y").year
+            curr_year = datetime.now().year
+            next_year = (datetime.now() + relativedelta(years=1)).year
+            if curr_year == year:
+                curr_year_value = curr_year_value + int(amount)
+            if next_year == year:
+                next_year_value = next_year_value + int(amount)
```
Comment (Safiyab) [inline robot/Cumulus/resources/RecurringDonationsPageObject.py:235]: Should this be if else ? I don't see how these both could be true at the same time

File: robot/Cumulus/resources/RecurringDonationsPageObject.py [hunk 2]
```diff
@@ -198,6 +209,35 @@ def get_next_payment_date_number(self, paynum):
...
+        validates them with the values displayed on the UI. """
+        installmentrow = npsp_lex_locators["erd"]["installment_row"]
+        installments = self.selenium.get_webelements(installmentrow)
+        count = len(installments)
+        print(f"Number of installments created is {count}")
+        i = 1
+        curr_year_value = 0
+        next_year_value = 0
+        values = {}
+        while i < count:
```
Comment (Safiyab) [inline robot/Cumulus/resources/RecurringDonationsPageObject.py:225]: what if count is 0 ? Should you be raising an exception for it ?

Comment (rjanjanam) [reply robot/Cumulus/resources/RecurringDonationsPageObject.py]: Thanks Safiya. Good point and added exception

File: robot/Cumulus/resources/RecurringDonationsPageObject.py (hunk 1 above)
Comment (rjanjanam) [reply robot/Cumulus/resources/RecurringDonationsPageObject.py]: Changed to else

File: robot/Cumulus/resources/RecurringDonationsPageObject.py [hunk 3]
```diff
@@ -87,7 +87,18 @@ def click_actions_button(self, button_name):
...
+            
+    def go_to_recurring_donation_related_opportunities_page(self,rd_id):
 
+        """ Navigates to the related opportunities page of the given recurring donation """
+        objectname = "npe03__Donations__r"
+        values =  self.npsp.get_url_formatted_object_name(objectname)
+        url = "{}/lightning/r/{}/related/{}/view".format(values['baseurl'],rd_id,objectname)
+        self.selenium.go_to(url)
+        self.salesforce.wait_until_loading_is_complete()
+        self.selenium.wait_until_location_contains("/view",timeout=60, message="Page not loaded")
```
Comment (rhakeshj) [inline robot/Cumulus/resources/RecurringDonationsPageObject.py:100]: Just curious to know if this always lets us know that the opportunity record has loaded successfully? Can we use any other data from the oppty itself to validate that the record is available to test?

Comment (skristem) [reply robot/Cumulus/resources/RecurringDonationsPageObject.py:100]: yup agree with rakesh, I recommend waiting for something to be visible on the page to confirm page has loaded and ready to test

File: robot/Cumulus/tests/browser/recurring_donations/zedit_erd_day_of_month_with_manual_opportunity.robot [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+Setup Test Data
+        [Documentation]     Data setup needed for the testcase. Creates a recurring
+        ...                 donation of type open linked to a contact using backend API
+        ${NS} =             Get NPSP Namespace Prefix
+        Set Suite Variable  ${NS}
+
+        #Create a Recurring Donation
+        &{contact1_fields}=   Create Dictionary                     Email=rd2tester@example.com
+        &{recurringdonation_fields} =	Create Dictionary           Name=ERD With Manual Donation
```
Comment (skristem) [inline robot/Cumulus/tests/browser/recurring_donations/zedit_erd_day_of_month_with_manual_opportunity.robot:25]: adjust the spacing

File: robot/Cumulus/tests/browser/recurring_donations/zedit_erd_day_of_month_with_manual_opportunity.robot [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+       Click Link                              link=Edit
+       Wait until Modal Is Open
+       Select Value From Dropdown              Stage                          ${stage}
+       Click Modal Button                      Save
+
+Create An Opportunity Related to Recurring Donation
+       [Documentation]         Create a manual related opportunity from the
+       ...                     Recurring Donation's opportunity related list
+       [Arguments]                       ${rd_id}                            ${Stage_Type}
+       go to recurring donation related opportunities page                   ${rd_id}
```
Comment (skristem) [inline robot/Cumulus/tests/browser/recurring_donations/zedit_erd_day_of_month_with_manual_opportunity.robot:62]: plz change the casing to our standard format

**Pull request merged**